src/main/kotlin/com/demonwav/mcdev/util/McBytecodeUtil.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+@file:JvmName("McBytecodeUtil")
+package com.demonwav.mcdev.util
+
+import com.intellij.psi.PsiArrayType
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiClassType
+import com.intellij.psi.PsiField
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.PsiPrimitiveType
+import com.intellij.psi.PsiType
+import com.intellij.psi.util.TypeConversionUtil
+
+private const val INTERNAL_CONSTRUCTOR_NAME = "<init>"
+
+// Type
+
+internal val PsiPrimitiveType.internalName: Char
+    get() = when (this) {
+        PsiType.BYTE -> 'B'
+        PsiType.CHAR -> 'C'
+        PsiType.DOUBLE -> 'D'
+        PsiType.FLOAT -> 'F'
+        PsiType.INT -> 'I'
+        PsiType.LONG -> 'J'
+        PsiType.SHORT -> 'S'
+        PsiType.BOOLEAN -> 'Z'
+        PsiType.VOID -> 'V'
+        else -> throw IllegalArgumentException("Unsupported primitive type: $this")
+    }
+
+private fun PsiClassType.erasure() = TypeConversionUtil.erasure(this) as PsiClassType
+
+internal val PsiClassType.internalName
+    get() = erasure().resolve()!!.internalName
```

Comment:
> The issue with your assumption is uncaught exceptions in IntelliJ don't fail silently. A bug notification pops up, with the stacktrace, and information on where the exception originated (this plugin).

That is correct, and once we have such an exception we can specifically modify the code to handle the situations in which the `resolve` method can return null.

> If it returns null then just don't append to the builder and return it? Either that, or throw a specific exception so the actual failure can be handled, don't just throw a NPE (or cause an NPE to be thrown).

Returning without appending anything would be a bad idea because you would run into many seemingly unrelated errors in the code because the result of these methods is often used for comparison of two method signatures. I agree that a NPE is not a good solution because of its ambiguity but I believe it is a good temporary solution until we can properly handle these situations.

This is essentially one case were I had to keep the code simple by making certain assumptions (e.g. here: that it will always `resolve` correctly) because other features took much longer than expected because of several special cases I had to handle. This is one of the parts I'd like to improve but I won't have much time for this in the next weeks.